src/vs/platform/terminal/common/capabilities/commandDetectionCapability.ts

```diff
@@ -197,6 +198,34 @@ export class CommandDetectionCapability extends Disposable implements ICommandDe
 		this._promptInputModel.setContinuationPrompt(value);
 	}
 
+	private _clearDecorationsInViewport(): void {
+		// Reset the current command's decorations by disposing markers
+		this._currentCommand.commandStartMarker?.dispose();
+		this._currentCommand.commandStartMarker = undefined;
+
+		// Above two line was enough to remove stale decorations.
+		// Should we still be disposing other "related"
+		this._currentCommand.commandExecutedMarker?.dispose();
+		this._currentCommand.commandExecutedMarker = undefined;
+
+		this._currentCommand.promptStartMarker?.dispose();
+		this._currentCommand.promptStartMarker = undefined;
+
+		this._currentCommand.currentContinuationMarker?.dispose();
+		this._currentCommand.currentContinuationMarker = undefined;
+
+		this._currentCommand.commandFinishedMarker?.dispose();
+		this._currentCommand.commandFinishedMarker = undefined;
+
+		// the lists goes on with commandLines..commandRightPromptStartX and commandRightPromptEndX... etc
+
+		// Should we be removing all markets in commandMarkers?
+		for (const marker of this._commandMarkers) {
```

Comment:
this seems dangerous, but thought it make sense since pre-existing `this._clearCommandsInViewport();` gets called before this function I made.